packages/site/cms-site/src/cookies/useOneTrustCookieApi.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+
+import { CookieApiHook } from "./CookieApiHook";
+
+type OneTrustOnConsentChangedEvent = {
+    detail: string[];
+    [key: string]: unknown;
+};
+
+type OnConsentChangedCallback = (event: OneTrustOnConsentChangedEvent) => void;
+
+type ConsentPayloadPurpose = {
+    Id: string;
+    TransactionType: "NO_CHOICE" | "CONFIRMED" | "OPT_OUT";
+};
+
+type ConsentGroup = {
+    OptanonGroupId: string;
+    PurposeId: string;
+    [key: string]: unknown;
+};
+
+type DomainData = {
+    ConsentIntegrationData: {
+        consentPayload: {
+            purposes: ConsentPayloadPurpose[];
+        };
+        [key: string]: unknown;
+    };
+    Groups: ConsentGroup[];
+    [key: string]: unknown;
+};
+
+type OneTrust = {
+    ToggleInfoDisplay: () => void;
+    OnConsentChanged: (callback: OnConsentChangedCallback) => void;
+    GetDomainData: () => DomainData;
+    [key: string]: unknown;
+};
+
+type WindowWithOneTrust = Window & {
+    OneTrust: OneTrust;
+};
+
+const isWindowWithOneTrust = (window: Window): window is WindowWithOneTrust => {
+    return "OneTrust" in window;
+};
+
+export const useOneTrustCookieApi: CookieApiHook = () => {
+    const [consentedCookies, setConsentedCookies] = React.useState<string[]>([]);
+
+    const startListeningForConsentChanges = React.useCallback((oneTrust: OneTrust) => {
```

Comment:
Optional: You could defined this function in the useEffect, then you don't need a useCallback